openstf/model/split_energy.py

```diff
@@ -42,23 +44,68 @@ def split_energy(pid):
     error = components[["load", "Inschatting"]].diff(axis=1).iloc[:, 1]
     mae = error.abs().mean()
     coefdict.update({"MAE": mae})
+    coefsdf = convert_coefdict_to_coefsdf(pj, input_split_function, coefdict)
+
+    # Get the coefs of previous runs and check if new coefs are valid
+    last_coefsdict = db.get_energy_split_coefs(pj, mean=True)
+    last_coefsdf = convert_coefdict_to_coefsdf(pj, input_split_function, last_coefsdict)
+    false_key = are_new_coefs_valid(coefdict, last_coefsdict)
+    if false_key is not None:
+        # If coefs not valid, do not update the coefs in the db and send teams
+        # message that something strange is happening
+        monitoring.post_teams_alert(
+            f"New splitting coefficient {false_key} for pid {pj['id']} deviate strongly from previously stored coefficients."
+            + f"New coefficient: {coefdict[false_key]}, last coefficient: {last_coefsdict[false_key]}.",
+            coefsdf=coefsdf,
+        )
+        # Use the last known coefficients for further processing
+        return last_coefsdf
+    else:
+        # Save Results
+        db.write_energy_splitting_coefficients(coefsdf, if_exists="append")
+        logger.info(
+            "Succesfully wrote energy split coefficients to database", pid=pj["id"]
+        )
+        return coefsdf
+
+
+def are_new_coefs_valid(new_coefs, last_coefs):
+    """Check if new coefficients are valid.
 
-    # Get average coefs of previous runs
-    mean_coefs = db.get_energy_split_coefs(pj, mean=True)
-    # Loop over keys and check if the difference with the average value is not more than 100%
-    # In case the difference is more tha 100% of the average set KPI coefs as expected to False
+    Args:
+        new_coefs (dict): new coefficients for standard load profiles (i.e.
+            wind, solar, household)
+        mean_coefs (dict): average of previous coefficients for standard load
```

Comment:
I think "mean_coefs" in the docstring should be changed to "last_coefs"

Reply:
Yes, changed.